Map/src/main/java/dk/sdu/se/f23/InVasion/map/MapPlugin.java

```diff
@@ -170,16 +178,20 @@ public void draw(GameData gameData) {
 
                     }
                 }
-
-
             }
 
-            mapFields.add(lines);
+        }
 
+    }
 
+    public void draw(GameData gameData) {
+        if(isClicked) {
+            EventDistributor.sendEvent(new BuyTowerEvent(new Point((int)ClickedField.getX(),(int)ClickedField.getY())), world);
```

Comment:
Hvad er tanken bag at rykke EventDistributor fra shopstate til map?

Reply:
shop do not know where the towers are XD

Reply:
i know its weird, but map knows where you click and then send an event, we could have an intermediate handling in shop, but that also seems weird